File: p_029_1_simpleactivityresult/src/main/java/com/startandroid/p_029_1_simpleactivityresult/MainActivity.java
package com.startandroid.p_029_1_simpleactivityresult;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    TextView tv;
    Button btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);

        tv = findViewById(R.id.tvName);
        btn = findViewById(R.id.btn);
        btn.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        Intent intent = new Intent(this, SecondActivity.class);
        startActivityForResult(intent, 1);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (data == null) {
            return;
        }
        String name = data.getStringExtra("name");
        tv.setText(tv.getText().toString() + name);
    }
}

File: k_094_service_kill_client/src/main/java/com/startandroid/k_094_service/MainActivity.kt
package com.startandroid.k_094_service

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        findViewById<Button>(R.id.btnStart).setOnClickListener {
            startService(Intent("MyService_first").putExtra("name", "value"))
        }
    }
}
//START_NOT_STICKY – сервис не будет перезапущен после того, как был убит системой

//START_STICKY – сервис будет перезапущен после того, как был убит системой

//START_REDELIVER_INTENT – сервис будет перезапущен после того, как был убит системой.
// Кроме этого, сервис снова получит все вызовы startService,
// которые не были завершены методом stopSelf(startId).